lib/circuitry/subscriber.rb

```diff
@@ -12,7 +12,7 @@ class Subscriber
     include Concerns::Async
     include Services::SQS
 
-    attr_reader :queue, :timeout, :wait_time, :batch_size, :lock, :ignore_visibility_timeout, :auto_delete
+    attr_reader :queue, :timeout, :wait_time, :batch_size, :lock, :ignore_visibility_timeout, :auto_delete, :before_message
```

Comment:
this is a clever way to get a handle of the message. We had a similar need to the deleter that could be cleaned up if it used the same method to get the message object.

Reply:
We made an upstream PR for the deleter workaround, but we should talk about reworking that to be unified and include this.